ruma-api-macros/src/api/response.rs

```diff
@@ -131,10 +131,17 @@ impl Response {
                         }
                     }
                     _ => quote! {
-                        resp_builder = resp_builder.header(
-                            #import_path::exports::http::header::#header_name,
-                            response.#field_name,
-                        );
+                        resp_builder.headers_mut()
+                            .map(|map| {
+                                match map.entry(#import_path::exports::http::header::#header_name) {
```

Comment:
I've had a look at `HeaderMap`s API: It doesn't look like we need to use the entry API, there's just an `insert` that does what we want.